mkosi/qemu.py

```diff
@@ -324,11 +325,23 @@ def run_qemu(args: MkosiArgs, config: MkosiConfig) -> None:
             if kernel:
                 cmdline += ["-kernel", kernel]
 
-            cmdline += ["-append", " ".join(config.kernel_command_line + config.kernel_command_line_extra)]
+            if config.output_format == OutputFormat.disk:
+                # We can't rely on gpt-auto-generator when direct kernel booting so synthesize a root=
+                # kernel argument instead.
+                root = finalize_root(find_partitions(fname))
+                if not root:
+                    die("Cannot direct boot image without root or usr partition")
```

Comment:
direct**ly**

Reply:
That grammatical error is grating. "Cannot perform a direct kernel boot", or "Direct kernel boot is not possible", etc.